src/core/public/chrome/ui/header/header_search_bar.tsx

```diff
@@ -200,50 +207,52 @@ export const HeaderSearchBar = ({ globalSearchCommands, panel, onSearchResultCli
       })}
       isLoading={isLoading}
       aria-label="Search the menus"
-      data-test-subj="search-input"
+      data-test-subj="global-search-input"
       className="searchInput"
       onFocus={() => {
-        const inputEl = inputElRef.current;
-        if (inputEl) {
-          const width = inputEl.getBoundingClientRect().width;
-          setPanelWidth(width);
-        }
+        setIsPopoverOpen(true);
       }}
     />
   );
 
+  const searchBarPanel = (
+    <EuiPanel
+      hasBorder={false}
+      hasShadow={false}
+      paddingSize="none"
+      data-test-subj="search-result-panel"
+    >
+      <EuiFlexGroup direction="column" gutterSize="s">
+        <EuiFlexItem>{searchBar}</EuiFlexItem>
+        <EuiFlexItem>{searchResultSections}</EuiFlexItem>
+      </EuiFlexGroup>
+    </EuiPanel>
+  );
+
   if (panel) {
-    return (
-      <EuiPanel
-        hasBorder={false}
-        hasShadow={false}
-        paddingSize="s"
-        data-test-subj="search-result-panel"
-      >
-        <EuiFlexGroup direction="column" gutterSize="s">
-          <EuiFlexItem>{searchBar}</EuiFlexItem>
-          <EuiFlexItem>{searchResultSections}</EuiFlexItem>
-        </EuiFlexGroup>
-      </EuiPanel>
-    );
+    return searchBarPanel;
   } else {
     return (
-      <EuiPopover
-        button={searchBar}
-        buttonRef={inputRef}
-        panelStyle={{ minWidth: panelWidth }}
-        zIndex={2000}
-        panelPaddingSize="s"
-        attachToAnchor={true}
-        ownFocus={false}
-        display="block"
-        isOpen={isPopoverOpen}
-        closePopover={() => {
-          setIsPopoverOpen(false);
-        }}
-      >
-        {searchResultSections}
-      </EuiPopover>
+      <>
+        {!isPopoverOpen && searchBar}
+        {isPopoverOpen && (
+          <EuiPopover
+            panelStyle={{ minWidth: '400px', minHeight: '100px' }}
+            button={<></>}
```

Comment:
Q: why using `<></>` as the popover anchor?

Reply:
Popover will always show with its anchor, the purpose of this will make it looks like only popover displayed. And button is required, so i put an empty element here.

![image](https://github.com/user-attachments/assets/118c07fb-8658-4f54-ae70-ca5dfab59a76)

Reply:
I see, thanks :)